### PR TITLE
Inji 408 error popup after maximum retries vc download

### PR DESCRIPTION
**Pull request by srikanth716**
Screenshot_1697533681

### CODE REVIEW

File: machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts [hunk 1]
```diff
@@ -173,7 +173,10 @@ export const ExistingMosipVCItemMachine =
                   actions: ['setMaxDownloadCount', 'setDownloadInterval'],
                 },
                 onError: {
-                  actions: log((_, event) => (event.data as Error).message),
+                  actions: [
+                    log((_, event) => (event.data as Error).message),
+                    'sendDownloadLimitExpireToVcMachine',
```
Comment (swatigoel) [inline machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts:178]: why do we have "Machine" in state name?
we can use "sendDownloadLimitExpire"

File: components/MessageOverlay.tsx [hunk 2]
```diff
@@ -46,6 +47,18 @@ export const MessageOverlay: React.FC<MessageOverlayProps> = props => {
               {props.message}
             </Text>
           )}
+          {props.failedVcs &&
```
Comment (swatigoel) [inline components/MessageOverlay.tsx:50]: Don't Pass VC related details here.
Text for MessageOverlay should be passed from MyVcsTab. MessgaeOverlay will just render it.

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts (hunk 1 above)
Comment (srikanth716) [reply machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts:178]: It is not an state, it was just an action that sends the event from ExistingMosipVCItemMachine.ts to Vc.ts. I will change the naming convention

File: components/MessageOverlay.tsx [hunk 3]
```diff
@@ -46,6 +46,7 @@ export const MessageOverlay: React.FC<MessageOverlayProps> = props => {
               {props.message}
             </Text>
           )}
+
```
Comment (swatigoel) [inline components/MessageOverlay.tsx:49]: remove this empty line

File: screens/Home/MyVcsTab.tsx [hunk 4]
```diff
@@ -77,6 +77,14 @@ export const MyVcsTab: React.FC<HomeScreenTabProps> = props => {
     }
   }, [controller.areAllVcsLoaded, controller.inProgressVcDownloads]);
 
+  let failedVCsList = [];
+  controller.downloadFailedVcs.forEach(v => {
```
Comment (swatigoel) [inline screens/Home/MyVcsTab.tsx:81]: rename `v` to `vc`

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: screens/Home/MyVcsTab.tsx [hunk 5]
```diff
@@ -233,6 +241,16 @@ export const MyVcsTab: React.FC<HomeScreenTabProps> = props => {
         buttonText={t('common:ok')}
         customHeight={'auto'}
       />
+
+      <MessageOverlay
+        isVisible={controller.isDownloadLimitExpires}
+        title={t('errors.downloadLimitExpires.title')}
+        message={downloadFailedVcsErrorMessage}
+        onButtonPress={controller.DELETE_VC}
```
Comment (swatigoel) [inline screens/Home/MyVcsTab.tsx:249]: As we are deleting VC on Ok button click, so make sure this popup is not dismissed on outside touch/click

**Review by swatigoel — APPROVED** [review]

**Pull request merged**